src/coreclr/vm/jitinterface.h

```diff
@@ -931,50 +1046,57 @@ protected :
 #endif
 
-#if defined(_DEBUG)
-    ULONG                   m_codeSize;     // Code size requested via allocMem
-#endif
-
-    size_t                  m_GCinfo_len;   // Cached copy of GCinfo_len so we can backout in BackoutJitData()
-    size_t                  m_EHinfo_len;   // Cached copy of EHinfo_len so we can backout in BackoutJitData()
-
-    ULONG32                 m_iOffsetMapping;
-    ICorDebugInfo::OffsetMapping * m_pOffsetMapping;
-
-    ULONG32                 m_iNativeVarInfo;
-    ICorDebugInfo::NativeVarInfo * m_pNativeVarInfo;
-
-    ICorDebugInfo::InlineTreeNode    *m_inlineTreeNodes;
-    ULONG32                           m_numInlineTreeNodes;
-    ICorDebugInfo::RichOffsetMapping *m_richOffsetMappings;
-    ULONG32                           m_numRichOffsetMappings;
-
 #ifdef FEATURE_ON_STACK_REPLACEMENT
     PatchpointInfo        * m_pPatchpointInfoFromJit;
     PatchpointInfo        * m_pPatchpointInfoFromRuntime;
     unsigned                m_ilOffset;
 #endif
 
-    // The first time a call is made to CEEJitInfo::GetProfilingHandle() from this thread
-    // for this method, these values are filled in.   Thereafter, these values are used
-    // in lieu of calling into the base CEEInfo::GetProfilingHandle() again.  This protects the
-    // profiler from duplicate calls to its FunctionIDMapper() callback.
-    struct GetProfilingHandleCache
+};
+
+#ifdef FEATURE_INTERPRETER
+class CInterpreterJitInfo final : public CEECodeGenInfo
+{
+public:
+    // ICorJitInfo stuff
+
+    CInterpreterJitInfo(MethodDesc* fd, COR_ILMETHOD_DECODER* header,
+                        EECodeGenManager* jm, bool allowInlining = true)
+        : CEECodeGenInfo(fd, header, jm, allowInlining)
     {
-        GetProfilingHandleCache() :
-            m_bGphIsCacheValid(false),
-            m_bGphHookFunction(false),
-            m_pvGphProfilerHandle(NULL)
+        CONTRACTL
         {
-            LIMITED_METHOD_CONTRACT;
-        }
+            NOTHROW;
+            GC_NOTRIGGER;
+            MODE_ANY;
+        } CONTRACTL_END;
+    }
 
-        bool                    m_bGphIsCacheValid : 1;        // Tells us whether below values are valid
-        bool                    m_bGphHookFunction : 1;
-        void*                   m_pvGphProfilerHandle;
-    } m_gphCache;
+    void allocMem(AllocMemArgs *pArgs) override;
+    void * allocGCInfo(size_t  size) override;
+    virtual void setEHcount (unsigned cEH) override;
+    virtual void setEHinfo (
+        unsigned      EHnumber,
+        const CORINFO_EH_CLAUSE* clause
+       ) override;
+
+    void WriteCodeBytes();
+    void WriteCode(EECodeGenManager * jitMgr) override;
 
+    void BackoutJitData(EECodeGenManager * jitMgr) override;
+    void SetDebugInfo(PTR_BYTE pDebugInfo) override;
+
+    void ResetForJitRetry()
+    {
+        CONTRACTL {
+            NOTHROW;
+            GC_NOTRIGGER;
+        } CONTRACTL_END;
+
+        CEECodeGenInfo::ResetForJitRetry();
```

Comment:
Do we want to do ASSERTE(false) here and/or an interpreter todo instead? I don't fully understand what jit retry implies or whether there is some scenario where the interpreter would want to do it, but it feels like something the interpreter wouldn't do.

It looks like jit retry is an altjit only thing, to begin with?

Reply:
This is a cleanup when jitting (even interpreter jitting) fails. It clears the state of the CEECodeGenInfo so that it can be reused.